platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDbDataSource.java

```diff
@@ -213,29 +182,15 @@ private static class MetadataFileFormatVersion {
     /** The range of valid leaf paths for data currently stored by this data source. */
     private volatile KeyRange validLeafPathRange = INVALID_KEY_RANGE;
 
-    private MerkleDbStatistics statistics;
-
-    /**
-     * When we register stats for the first instance, also register the global stats. If true then
-     * this is the first time stats are being registered for an instance.
-     */
-    private static final AtomicBoolean firstStatRegistration = new AtomicBoolean(true);
-
-    private final AtomicBoolean compactionEnabled = new AtomicBoolean();
-
-    /** When was the last medium-sized merge, only touched from single merge thread. */
-    private Instant lastMediumMerge;
-
-    /** When was the last full merge, only touched from single merge thread. */
-    private Instant lastFullMerge;
-
     /** Paths to all database files and directories */
     private final MerkleDbPaths dbPaths;
 
     private final AtomicBoolean closed = new AtomicBoolean(false);
 
-    /** A nanosecond-precise Clock */
-    private final NanoClock clock = new NanoClock();
+    /** Runs compactions for the storages of this data source */
+    final MerkleDbCompactionCoordinator compactor;
+
+    private MerkleDbStatisticsUpdater statisticsUpdater;
```

Comment:
Could this be final?

Reply:
Unfortunately, it can't be because we have method `MerkleDbDataSource#copyStatisticsFrom` that reassigns the updater. However, thanks to your comment, I found redundant construction of this object.